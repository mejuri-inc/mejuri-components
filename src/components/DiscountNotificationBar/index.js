import React from 'react'
import PropTypes from 'prop-types'
import { FormattedHTMLMessage } from 'react-intl'
import GenericNotificationBar from 'components/GenericNotificationBar'

const linkUrl = (host, url) => `${host}${url}`

const DiscountNotificationBar = ({ apiHost, name, link }) => {
  if (!name) return null
  return (
    <GenericNotificationBar>
      <FormattedHTMLMessage
        id='header.notifications.discount'
        values={{
          name,
          link: linkUrl(apiHost, link)
        }}
      />
    </GenericNotificationBar>
  )
}

DiscountNotificationBar.defaultProps = {
  apiHost: typeof window !== 'undefined' ? window.location.origin : '',
  link: '/static/crewfineprint'
}

DiscountNotificationBar.propTypes = {
  apiHost: PropTypes.string,
  name: PropTypes.string,
  link: PropTypes.string
}

export default DiscountNotificationBar
