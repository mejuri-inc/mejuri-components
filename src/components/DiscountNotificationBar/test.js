/* eslint-disable no-unused-expressions */
import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'themes/styled'
import { IntlProvider } from 'react-intl'

import DiscountNotificationBar from './index'

describe('DiscountNotificationBar', () => {
  it('Renders nothing if no name', () => {
    const { container } = render(<DiscountNotificationBar />)
    expect(container).toBeNull
  })

  it('Renders just fine if a name is provided', () => {
    const { container } = render(
      <IntlProvider
        locale='en'
        messages={{
          'header.notifications.discount': 'This is a notification {name}'
        }}
      >
        <ThemeProvider theme={Theme}>
          <DiscountNotificationBar name='Roger Rabbit' />
        </ThemeProvider>
      </IntlProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
