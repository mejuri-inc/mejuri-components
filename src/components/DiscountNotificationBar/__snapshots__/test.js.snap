// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountNotificationBar Renders just fine if a name is provided 1`] = `
<div>
  <div
    class="sc-bdnylx ANRse"
  >
    <span>
      This is a notification Roger Rabbit
    </span>
    <a
      href="http://localhost/static/crewfineprint"
    >
      <span>
        The Fine Print
      </span>
    </a>
  </div>
</div>
`;
